source/gamemap/GameMap.cpp

```diff
@@ -905,68 +907,122 @@ void GameMap::createAllEntities()
 {
     mTileSet = ConfigManager::getSingleton().getTileSet(mTileSetName);
 
-    std::vector<GameEntity*> entities;
-    // Create OGRE entities for map tiles
-    for (int jj = 0; jj < getMapSizeY(); ++jj)
+    for (Creature* creature : mCreatures)
     {
-        for (int ii = 0; ii < getMapSizeX(); ++ii)
-        {
-            getTile(ii,jj)->createMesh();
-        }
+        //Set up definition for creature. This was previously done in createMesh for some reason.
+        creature->setupDefinition(*this, *ConfigManager::getSingleton().getCreatureDefinitionDefaultWorker());
+        //Doesn't do anything currently.
+        //creature->restoreInitialEntityState();
     }
 
-    // Create OGRE entities for rendered entities
-    for (RenderedMovableEntity* rendered : mRenderedMovableEntities)
+    if(isServerGameMap())
     {
-        rendered->createMesh();
-        rendered->setPosition(rendered->getPosition(), false);
-        entities.push_back(rendered);
-    }
+        // Set positions and update active spots
+        for (RenderedMovableEntity* rendered : mRenderedMovableEntities)
+        {
+            rendered->setPosition(rendered->getPosition(), false);
+        }
 
-    // Create OGRE entities for the creatures
-    for (Creature* creature : mCreatures)
-    {
-        creature->createMesh();
-        creature->setPosition(creature->getPosition(), false);
-        entities.push_back(creature);
-    }
+        for (Room* room : mRooms)
+        {
+            room->updateActiveSpots();
+        }
+
+        for (Spell* spell : mSpells)
+        {
+            spell->setPosition(spell->getPosition(), false);
+        }
 
-    // Create OGRE entities for the map lights.
-    for (MapLight* mapLight: mMapLights)
+        for (Trap* trap : mTraps)
+        {
+            trap->updateActiveSpots();
+        }
+
+        for (Creature* creature : mCreatures)
+        {
+            //Set up definition for creature. This was previously done in createMesh for some reason.
+            creature->setupDefinition(*this, *ConfigManager::getSingleton().getCreatureDefinitionDefaultWorker());
```

Comment:
It's called once for each creature, as it has not been called for creatures that have been loaded from the map.